src/redis-benchmark.c

```diff
@@ -605,17 +622,24 @@ static void writeHandler(aeEventLoop *el, int fd, void *privdata, int mask) {
     /* Initialize request when nothing was written. */
     if (c->written == 0) {
         /* Enforce upper bound to number of requests. */
-        int requests_issued = 0;
-        atomicGetIncr(config.requests_issued, requests_issued, config.pipeline);
-        if (requests_issued >= config.requests) {
-            return;
+        if (config.qps == 0) {
+            int requests_issued = 0;
+            atomicGetIncr(config.requests_issued, requests_issued, config.pipeline);
+            if (requests_issued >= config.requests) {
+                return;
+            }
         }
 
         /* Really initialize: randomize keys and set start time. */
         if (config.randomkeys) randomizeClientKey(c);
         if (config.cluster_mode && c->staglen > 0) setClusterKeyHashTag(c);
         atomicGet(config.slots_last_update, c->slots_last_update);
-        c->start = ustime();
+        if (config.qps) {
+            long long timestamp = popItem();
+            if (timestamp == -1) return;
+            else  c->start = timestamp;
```

Comment:
I think there's a problem here. This timestamp is not the actual timestamp sent to the Server.

Reply:
yes, I have modified the code to use the actual timestamp.